components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/impl/BackendJWTAuthenticationImpl.java

```diff
@@ -0,0 +1,119 @@
+ * under the License.
+ */
+
+package org.wso2.carbon.apimgt.rest.api.util.impl;
+
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.cxf.message.Message;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.common.gateway.dto.JWTValidationInfo;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+import org.wso2.carbon.apimgt.impl.RESTAPICacheConfiguration;
+import org.wso2.carbon.apimgt.impl.jwt.SignedJWTInfo;
+import org.wso2.carbon.apimgt.impl.utils.APIUtil;
+import org.wso2.carbon.apimgt.rest.api.common.RestApiConstants;
+import org.wso2.carbon.apimgt.rest.api.util.MethodStats;
+import org.wso2.carbon.apimgt.rest.api.util.authenticators.AbstractOAuthAuthenticator;
+import org.wso2.carbon.apimgt.rest.api.util.utils.OauthUtils;
+import java.text.ParseException;
+
+/**
+ * This class is for Authenticate API requests that coming with X-JWT-Assertion header. X-JWT-Assertion header transporting
+ * the backend JWT.Validating the JWT against the token issuer is not required since the token coming from well known party
+ * Un-authorize access are denied by network policies.
+ * */
+
+public class BackendJWTAuthenticationImpl extends AbstractOAuthAuthenticator {
+
+    private static final Log log = LogFactory.getLog(BackendJWTAuthenticationImpl.class);
+    private boolean isRESTApiTokenCacheEnabled;
+
+    @Override
+    public boolean authenticate(Message message) throws APIManagementException {
+        RESTAPICacheConfiguration cacheConfiguration = APIUtil.getRESTAPICacheConfig();
+        isRESTApiTokenCacheEnabled = cacheConfiguration.isTokenCacheEnabled();
+        String accessToken = (String) message.get(RestApiConstants.JWT_TOKEN);
+
+        if ((accessToken != null && StringUtils.countMatches(accessToken, APIConstants.DOT) != 2)) {
+            log.error("Invalid JWT token. The expected token format is <header.payload.signature>");
+            return false;
+        }
+        try {
+            SignedJWTInfo signedJWTInfo = null;
+            String token = (String) message.get(RestApiConstants.JWT_TOKEN);
+            if (accessToken != null) {
+                signedJWTInfo = getSignedJwt(accessToken);
+            }
+
+            if (signedJWTInfo != null) {
+                String jwtTokenIdentifier = getJWTTokenIdentifier(signedJWTInfo);
+
+                JWTValidationInfo jwtValidationInfo = new JWTValidationInfo();
+                jwtValidationInfo.setValid(true);
+                if (isRESTApiTokenCacheEnabled) {
+                    getRESTAPITokenCache().put(jwtTokenIdentifier, jwtValidationInfo);
+                }
+                //Validating scopes
+                OauthUtils oauthUtils = new OauthUtils();
+                return oauthUtils.handleScopeValidation(message, signedJWTInfo, token);
```

Comment:
Lets change the class name and make this a static method.
```suggestion
                return JWTUtil.handleScopeValidation(message, signedJWTInfo, token);
```